torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -147,8 +135,7 @@ TORCH_LIBRARY(c10d, m) {
   // enable
   // __torch_dispatch__.
   m.def(
-      "broadcast_",
-      dispatch(c10::DispatchKey::CompositeExplicitAutograd, broadcast_));
```

Comment:
Hey @bdhirsh, in the comment above, Jiewen mentioned "It's important to register the op to the CompositeExplicitAutograd key to enable __ torch_dispatch __.". 

With this new change we are not specifying a default implementation or the CompositeExplicitAutograd dispatch key, is this okay? Will __ torch_dispatch __ still work?

Reply:
tldr is yep, it should still work fine.

(I think the thing Jiewen was referring to is that if you made your kernel `CompositeImplicitAutograd`, it wouldn't work with `__torch_dispatch__`. You're writing separate CPU + CUDA implementations though, which will work fine)

Reply:
@H-Huang nit: shall we update the comment here given @bdhirsh's comment?

Reply:
Yep sounds good